packages/batch-submitter/src/batch-submitter/batch-submitter.ts

```diff
@@ -149,7 +149,13 @@ export abstract class BatchSubmitter {
     }
   }
 
-  protected _shouldSubmitBatch(batchSizeInBytes: number): boolean {
+  protected _shouldSubmitBatch(
+    batchSizeInBytes: number,
+    numberOfElements: number
+  ): boolean {
+    if (numberOfElements === this.maxBatchSize) {
+      return true
+    }
     const currentTimestamp = Date.now()
     if (batchSizeInBytes < this.minTxSize) {
```

Comment:
Perhaps `batchSizeInBytes < this.minTxSize && numberOfElements < this.maxBatchSize`?

Reply:
Yep! Exactly this ^ 

Good catch